integration/ec2_test.go

```diff
@@ -269,7 +270,7 @@ func TestIAMNodeJoin(t *testing.T) {
 	// the proxy should eventually join the cluster and heartbeat
 	require.Eventually(t, func() bool {
 		proxies, err := authServer.GetProxies()
-		require.NoError(t, err)
+		assert.NoError(t, err)
```

Comment:
Hm, I could be wrong, but I'm not sure the behaviour this results in is good. 

If this `GetProxies` call produces an error once, then `Errorf` will be invoked on the `testing.T`, but it will be retried and the test will continue. At the end of the test, despite the retry succeeding, surely this results in the test showing as "failed" - I can't see a point in continuing the test if it's going to result in a failure state anyway ?

I'd think for the behaviour we want, we'd probably want to use [`EventuallyWithT`](https://pkg.go.dev/github.com/stretchr/testify/require#EventuallyWithT) to intercept the assertion failures during the retries.

Reply:
Good call. I never understood `EventuallyWithT` until now!